src/mds/SessionMap.h

```diff
@@ -418,6 +420,7 @@ class Session : public RefCountedObject {
   MDSAuthCaps auth_caps;
 
   xlist<Session*>::item item_session_list;
+  xlist<Session*>::item item_broken_root_squash_clients;
```

Comment:
I'm not sure it's justified to add another maintained xlist, along with its corresponding item in the `Session`, for what can be easily calculated ad-hoc given a session map when generating the metrics for a beacon. Am I right to assume that we will have << 100 concurrent sessions in most cases?

Reply:
If we still want to optimize this, we'll achieve 80% of the benefit with a simple flag set on the session map if a problematic client is seen in `handle_client_session`, which will enable scanning in the metric collection code and will be reset if no broken clients are detected.

Reply:
> I'm not sure it's justified to add another maintained xlist, along with its corresponding item in the `Session`, for what can be easily calculated ad-hoc given a session map when generating the metrics for a beacon.

We can know at session open if there is a problem. There is no reason to calculate it every beacon. My other preference for managing this is a `std::set<Session*>` which should typically be empty. I think I'll change it to that.

> Am I right to assume that we will have << 100 concurrent sessions in most cases?

Some clusters can have several thousand.

Reply:
> We can know at session open if there is a problem. There is no reason to calculate it every beacon.

In that case, how about a single counter, maintained in the session handlers?